### PR TITLE
fix: vcfsplit BND record ordering

### PR DESCRIPTION
**Pull request by tedil**
This PR solves #229 by keeping a `BND event id or mate ids → output writer` mapping. That means that the first record of a BND is written to the current chunk and all subsequent records that are encountered during iteration that belong to the same event / have the same set of mate ids get written to that same chunk.
(I am not quite sure if I accidentally changed old behaviour by oversimplifying the code)

### CODE REVIEW

File: src/bcf/split.rs [hunk 1]
```diff
@@ -136,79 +74,33 @@ impl BreakendGroup {
...
+        } else {
+            None
+        }
     }
-}
 
-fn mateids(record: &mut bcf::Record) -> Option<Vec<&[u8]>> {
-    match record.info(b"MATEID").string() {
-        Ok(Some(s)) => Some(s.clone().into_iter().collect_vec()),
-        _ => None,
```
Comment (tedil) [inline src/bcf/split.rs:212]: Note the difference to the new code which goes from `&[u8]` to `Vec<u8>`; the slice seems to have been "unstable" due to rust_htslib in the background so would sometimes be invalid (i.e. the memory it was pointing to changed!), which is why we allocate now.

File: src/bcf/split.rs [hunk 2]
```diff
@@ -92,36 +56,10 @@ enum BreakendGroup {
...
-    fn from(rec: &mut bcf::Record) -> Option<Self> {
-        if let Some(event) = event(rec) {
+    fn from(rec: &bcf::Record) -> Option<Self> {
+        if let Some(event) = rec.event() {
             Some(BreakendGroup::Event(event))
-        } else if let Some(mateids) = mateids(rec) {
+        } else if let Some(mateids) = rec.mateids() {
             let mut mates: Vec<_> = mateids
                 .into_iter()
                 .map(|mateid| mateid.to_owned())
```
Comment (johanneskoester) [inline src/bcf/split.rs:65]: I think the entire collect here can now be replaced by a simple clone of the Vec.

Comment (johanneskoester) [reply src/bcf/split.rs:65]: See clippy warning.

**Review by johanneskoester — CHANGES_REQUESTED** [review]

File: src/bcf/split.rs (hunk 2 above)
Comment (tedil) [reply src/bcf/split.rs]: Yup! (I have in-IDE on-the-fly clippy lints enabled, but this somehow didn't show up)

**Pull request merged**